### DISCUSSION TITLE
High-state error  so-docker-prune Python 3.6

### DISCUSSION THREAD

**Discussion by sleepingbel** [Security-Onion-Solutions/securityonion#4703] [Unsupported Versions]
Hello All

Standalone install version 2.3.40 upgraded to 2.3.60

I have this error when i run high-state

![Knipsel](https://user-images.githubusercontent.com/29328171/124505409-bd5bb100-ddc9-11eb-92cb-840817866f96.PNG)

It is the same error as in this topic 
[https://github.com/Security-Onion-Solutions/securityonion/discussions/4034](url)

Is there a solution ?

Thanks in advance
Bart

**Comment by rwwiv** [comment]
This error is unrelated to #4034, and it seems to be due to a docker prune being run in the background, or stuck running.

If you're able, I'd like to get some more information about what's going on to work on a solution:
* Does this error occur every time you try to run a highstate?
* Do you see any output from the command `ps aux | grep "docker.*prune" | grep -v "grep"` ?
* Have you tried rebooting the box and running a highstate once all the services have started?

**Reply by sleepingbel** [reply]
Hello William,

To answer your questions:

    Yes it happens every time i run highstate.
    There is no output for the grep
    Yes I rebooted the machine and runned again highstate with the same result
    I checked with docker ps the different docker stations and they are all version 2.3.60

I hope that this gives you the information you need

Regards
Bart

**Comment by facyber** [comment]
I solved this by identifying old docker images with `docker ps` and then removing them with `docker rm <CONTAINER_ID>` and I believe I also saw someone else recently also did this.

Do note, I can not guarantee that this is a good way to do it, but it worked for me and resolved issues. Someone with better knowledge can maybe confirm or correct my statement.

**Reply by Mastadamus** [reply]
I am getting that same error. I also realized no new data is showing in kibana or SOC since the upgrade last night. Will update IF I find out its related to that error.

**Reply by rwwiv** [reply]
@facyber, that is essentially what `so-docker-prune` is now doing before pruning images; it runs a docker container prune which removes all stopped containers. To be clear, are you removing stopped/exited containers using `docker ps -a` or are you seeing running containers still using old images?

`so-docker-prune` was seeing an issue before where old containers weren't being removed, which was causing it to fail to remove old images from the system. What @sleepingbel is seeing is a different error that's due to a docker prune of some kind already running when the highstate tries to run this script. Are you saying you saw that same error and were able to fix it by manually removing container?

**Reply by facyber** [reply]
Hey @rwwiv I guess I expressed badly. Indeed, I removed old images that are not in use anymore (not active), they all where for 2.3 version, while I had 2.5. For some reason they stuck on my setup, but removing them manually resolved the issue and no errors now.

If I remember correctly I did had same issue, because he/she refers to that other discussion where I commented on to. And that solution I read from someone else here in one of the threads but can't remember which one.

**Comment by sleepingbel** [comment]
Hello William,

To answer your questions:
1. Yes it happens every time i run highstate.
2. There is no output for the grep 
3. Yes I rebooted the machine and runned again highstate with the same result
4. I checked with docker ps the different docker stations and they are all version 2.3.60

I hope that this gives you the information you need

Regards 
Bart
